core-api/README.md

```diff
@@ -0,0 +1,102 @@
+interface is used for dispatching events to the Optimizely event end-point. Implementations of `dispatchEvent(LogEvent)` are expected
+to make an HTTP request of type `getRequestMethod()` to the `LogEvent#getEndpointUrl()` location. The corresponding request parameters and body
+are available via `LogEvent#getRequestParams()` and `LogEvent#getBody()` respectively.
+
+### NoopEventHandler
+[`NoopEventHandler`](https://github.com/optimizely/java-sdk/blob/master/core-api/src/main/java/com/optimizely/ab/event/NoopEventHandler.java)
+implements `EventHandler` with no side-effects. Useful for testing or non-production environments.
+
+## NotificationCenter
+[`NotificationCenter`](https://github.com/optimizely/java-sdk/blob/master/core-api/src/main/java/com/optimizely/ab/notification/NotificationCenter.java)
+is the centralized component for subscribing to notifications from the SDK. Subscribers must implement the `NotificationHandler<T>` interface
+and are registered via `NotificationCenter#addNotificationHandler`. Notifications are currently sent synchronously, so be sure that 
+implementations do not block unnecessarily.
+
+## ProjectConfig
+[`ProjectConfig`](https://github.com/optimizely/java-sdk/blob/master/core-api/src/main/java/com/optimizely/ab/config/ProjectConfig.java)
+represents the current state of the Optimizely project as configured through [optimizely.com](https://www.optimizely.com/).
+The interface is currently unstable and used only internally. All public access to these implementations are subject to change
+with each subsequent version.
+
+### DatafileProjectConfig
+[`DatafileProjecyConfig](https://github.com/optimizely/java-sdk/blob/master/core-api/src/main/java/com/optimizely/ab/config/DatafileProjectConfig.java)
+is an implementation of `ProjectConfig` backed by a file typically located within the Optimizely CDN.
+
+## ProjectConfigManager
+[`ProjectConfigManager`](https://github.com/optimizely/java-sdk/blob/master/core-api/src/main/java/com/optimizely/ab/config/ProjectConfigManager.java)
+is a factory class that serves a `ProjectConfig`. Implementations of this class serve to provide a consistent representation
+of a `ProjectConfig` that can be passed between services.
+
+### AtomicProjectConfigManager
+[`AtomicProjectConfigManager`](https://github.com/optimizely/java-sdk/blob/master/core-api/src/main/java/com/optimizely/ab/config/AtomicProjectConfigManager.java)
+is a static provider that can be updated atomically to provide a consistent view of a `ProjectConfig`.
+
+### PollingProjectConfigManager
+[`PollingProjectConfigManager`](https://github.com/optimizely/java-sdk/blob/master/core-api/src/main/java/com/optimizely/ab/config/PollingProjectConfigManager.java)
+ia a abstract class that provides the foundation for a dynamic factory that updates asynchronously within a background thread.
+Implementations of this class can be used to poll from a HTTP or file based datafile location.
```

Comment:
Change "a" to "an" and adding hyphens
...from an HTTP- or file-based datafile location.